Sync more changes from Joomla 1.7. (Mostly DS)

realityking opened the pull request.

eddieajau committed:
Code cleaning

eddieajau committed:
Fixed problem with Purge Expired Cache removing
administrator/cache/index.html file

Credit: Peter Chovancak, Nikolai Plath

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=25905

eddieajau committed:
Prevent setting more than 1 default home menu item per
menu.

Credit: Jean-Marie Simonet

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=26023

eddieajau committed:
Fixed embedded image converted to data in RSS feed : data URI
get's corrupted

Credit: Jeff Moll, Dennis Hermatski

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=25687

eddieajau committed:
Fixed module messages not displayed if they are defined before the
"jdoc:include message" tag in the template

Credit: Adrien BABORIER, Chris Davenport, Christophe Demko, Dennis
Hermatski

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=25675

eddieajau committed:
Fixed JHtml::tooltip quote problem

Credit: victor antoniak, Dennis Hermatski

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=25720

eddieajau committed:
Fixed Unable to run scriptfile on plg_editors-xtd type plugin

Credit: Michael Babker

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=25462

eddieajau committed:
Rename JVersion to JPlatform to avoid collion with CMS JVersion.

Commit:
Cleaned up After JVersion Rename

Rename of JVersion left the tests for it untouched. Renamed the test file and everything associated with JVersion to JPlatform, to catch up with change.

Commit:
Cleaning up after JDocument changes

Removing methods under test while ignoring the tests
caused a mess that needed cleaned up.

Commit:
Fixed JStringTest

Darwin bug was causing tests to fail. Added test for Darwin, and if
tests are running under Darwin, it marks the international string
tests as skipped, and gives the reason why.

Commit:
Fixed broken JCacheStorageTest test

Test case did not handle mimimum cache life properly. Also moved
test into the JoomlaTestCase class hierarchy because of its dependencies
and cleaned up duplicated code.

Commit:
Fixed JCacheTest cases

JCacheTest test cases were pointed at a directory structure that no
longer exists in the new testing environment. I pointed them to the
correct location.

Commit:
JDocumentRenderer tests skipped

The JDocumentRendererAtom and JDocumentRendererRSS classes are too tightly coupled
for good testing right now, so I marked the tests as skipped with a todo to come back after
breaking more dependencies in the code.

eddieajau committed:
Added new feature [#25732] Add JFormRuleOptions rule to validate
type="list" parameters

Credit: Mark Dexter

realityking committed:
Add JHtmlBehavior::multiselect().

realityking committed:
Update FancyUpload to version 3.0.1.

realityking committed:
Merge branch 'master' into at

Conflicts:
	tests/suite/joomla/document/JDocumentTest.php

eddieajau committed:
Merge pull request #31 from Paladin/JVersion

Cleaned up After JVersion Rename

chdemko committed:
rollback menu

chdemko committed:
Corrects some errors in the router + add some unit test

chdemko committed:
Improve JRegistry

GeraintEdwards committed:
Edited libraries/joomla/database/databasequery.php via GitHub

GeraintEdwards committed:
Edited libraries/joomla/application/component/modellist.php via GitHub

LouisLandry committed:
Merge remote branch 'elin/docblocks'

Conflicts:
	libraries/joomla/environment/uri.php


Signed-off-by: Louis Landry <louis.landry@joomla.org>

LouisLandry committed:
Merge pull request #32 from Paladin/JDocument

Cleaning up after JDocument changes

LouisLandry committed:
Merge pull request #37 from Paladin/JDocumentRenderer

JDocumentRenderer tests skipped

LouisLandry committed:
Merge pull request #35 from Paladin/JCache

Fixed JCacheTest cases.  Thanks Arlen!

LouisLandry committed:
Merge pull request #34 from Paladin/JCacheStorage

Fixed broken JCacheStorageTest test

LouisLandry committed:
Merge pull request #33 from Paladin/JString

Fixed JStringTest

LouisLandry committed:
Merge pull request #39 from realityking/uploader

Update FancyUpload to version 3.0.1.

LouisLandry committed:
Merge pull request #12 from realityking/at

New Unit tests

Commit:
Cleared Fatal crash if JTestConfig not found

Added different config file to JDatabaseMySQLTest to allow an independent decision
by the tester on whether these tests should be run. It now requires
config_mysql.php to contain the JMySQLTestConfig class to be present or the tests
will be skipped.

realityking committed:
Move JUpdater:arrayUnique() to JArrayHelper::arrayUnique(). Keep JUpdater:arrayUnique() as an alias an deprecate it.

realityking committed:
Merge branch 'master' into opensearch

LouisLandry committed:
Merge pull request #45 from Paladin/JDatabaseMySQL

Cleared Fatal crash if JTestConfig not found

LouisLandry committed:
Merge pull request #18 from realityking/opensearch

Add JDocumentOpensearch.

GeraintEdwards committed:
Put the getter in the wrong class.

GeraintEdwards committed:
Edited libraries/joomla/database/databasequery.php via GitHub

LouisLandry committed:
Merge remote branch 'louis/master'

realityking committed:
Update MooTools to version 1.3.2.

realityking committed:
Update SqueezeBox to version 1.2

realityking committed:
Namespace the checkAll() function

LouisLandry committed:
Merge pull request #49 from realityking/trunk

Sync up media changes from trunk

LouisLandry committed:
Merge pull request #46 from realityking/array

Move JUpdater:arrayUnique() to JArrayHelper::arrayUnique().

LouisLandry committed:
Merge pull request #47 from GeraintEdwards/master

2 platform changes that will enable Joomfish to interact with database queries more effectively (redux)

LouisLandry committed:
Merge pull request #50 from realityking/checkall

Namespace the checkAll() function

LouisLandry committed:
Merge pull request #41 from chdemko/master

Component install rollback function rollback_menu is unimplemented leaving items in jos_menu

LouisLandry committed:
Merge pull request #42 from chdemko/router

Corrects some errors in the router + add some unit test

sebastianbergmann committed:
Rename configuration following PHPUnit best practices.

sebastianbergmann committed:
Attempt at skipping tests that depend on the database when none is configured/available.

sebastianbergmann committed:
Ignore tests/test_application/cache.

svanschu committed:
fix removeUserFromGroup()

LouisLandry committed:
Merge pull request #51 from sebastianbergmann/master

Test suite cleanup

elinw committed:
Let getModules get the module parameters.

realityking committed:
Use local storage for tabs

realityking committed:
Don't render meta elements without content

LouisLandry committed:
Merge pull request #56 from realityking/tabs

Use local storage for tabs

LouisLandry committed:
Merge pull request #55 from realityking/meta

Don't render meta elements without content

LouisLandry committed:
Merge pull request #52 from lorddarkangel/master

fix for removeUserFromGroup(), Ticket does not exist for it

LouisLandry committed:
Merge pull request #38 from realityking/multiselect

Add JHtmlBehavior::multiselect() and move the multiselect JavaScript code

LouisLandry committed:
Merge pull request #26 from mbabker/master

Add batch processing to framework (Feature Request #22590)

LouisLandry committed:
Merge pull request #53 from elinw/menuhelper

Let getModules get the module parameters.

Commit:
Restoring memory limit after daemon test

eddieajau committed:
Merge remote branch 'origin/master'

eddieajau committed:
Remove .gitignore from the branch. Added README with instructions on
paths to add to .gitignore.

Commit:
Adding files for continuous integration

eddieajau committed:
Code cleaning.

eddieajau committed:
Added feature [#24930] Add to debug log information component and
module start time and memory use when showing (Vladimir Serotyukov,
Dennis Hermatski).

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=24930

eddieajau committed:
Fixed issue [#23165] Make index.html's valid (Rouven Weßling).

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=23165

eddieajau committed:
Fixed issue [#26104] JModelForm validate method is missing group
option (Roland Dalmulder).

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=26104

eddieajau committed:
Code cleaning.

eddieajau committed:
Code cleaning.

eddieajau committed:
Code cleaning.

LouisLandry committed:
Merge pull request #58 from ianmacl/ci

Adding files for continuous integration

LouisLandry committed:
Merge pull request #57 from ianmacl/ini_restore

Restoring memory limit after daemon test

Commit:
Workaround for phpcs issue

Commit:
Turning phpcs output on

elinw committed:
applied patch

Commit:
Fixing build issue which was scrapping reports

Commit:
Continue build on error

realityking committed:
Add JRequest::getUInt() to allow filtering for unsigned integers.

realityking committed:
Escape more than just ampersands in JRoute.

Commit:
Added sample phpmd rule file

LouisLandry committed:
Merge pull request #59 from elinw/docblocks

Docblocks

LouisLandry committed:
Merge pull request #60 from realityking/secruity

Secruity

rcorral committed:
Rearranged the if statement

It will be quicker to check for null everytime than the $plugin variable

rcorral committed:
Significantly improved perfornce of loading plugins

Without having the need to call self::getPlugin() every time a specific plugin or a plugin type is loaded, it significantly improves performance as we won't be looping through the same plugin array that we are already going through. I tested this with 2000 plugins, the performance going from about ~6 seconds to ~1.4 seconds.
I wrapped the call on an if statement that just checks to see if params is defined, if it is not then we call it, but we never go into this if statement during a normal execution for Joomla, I just left it in there just in case, but if possible I suggest removing it completely.
I will be committing my test cases within the next day or two to the SVN as I am trying to improve the performance of the plugin API.

LouisLandry committed:
Merge pull request #61 from rcorral/master

Rearranged if statement

elinw committed:
Deprecate JException

LouisLandry committed:
Merge pull request #62 from elinw/docblocks

Deprecate JException

elinw committed:
Docblocks mysdl

elinw committed:
Docblocks for some driver files.

elinw committed:
database/database/ doc blocks work

LouisLandry committed:
Some minor fixes to unit tests.

Commit:
Merge github.com:joomla/joomla-platform into ci

Commit:
Merge pull request #64 from LouisLandry/master

A couple minor adjustments to the JApplication constructor and JDocumentOpenSearch tests.

Commit:
Merge github.com:joomla/joomla-platform into ci

elinw committed:
copy editing

elinw committed:
Thanks Ian

Commit:
Merge pull request #65 from elinw/docblocks

Docblocks

LouisLandry committed:
Merge pull request #25 from klas/master

Fixes for #22587 and #24285

elinw committed:
Docblocks in database continued

elinw committed:
Tablenested docblocks

elinw committed:
More work on database docblocks, mainly tables

Change-Id: I714c62c3ccc7bd0ec49d72d3af10eca0d94697a2

elinw committed:
More docblocks, mainly in the tables.

elinw committed:
Doc block work, a missing index.html file, some comment corrections,
mainly in Document.

elinw committed:
Docblocks in archive

elinw committed:
File system doc blocks.

Commit:
Minor change made to JDate::setTimezone to correct PHP Strict Error

Commit:
Remove JButton::fetchID Parent Class which is inconsistent with all child classes and not used in core - produces PHP Strict Error Message

LouisLandry committed:
Merge pull request #70 from AmyStephen/master

PHP Strict Errors

LouisLandry committed:
Merge pull request #68 from elinw/docblocks

Docblocks

LouisLandry committed:
Some necessary adjustments to JCli

LouisLandry committed:
Fixing an error in the JHtml test where a directory is being created
when it already exists.  These tests need to be seriously looked over at
some point but this is a bandaid.

LouisLandry committed:
Some minor modifications to JURI and associated tests as well as a new
README.

LouisLandry committed:
Merge pull request #72 from LouisLandry/master

Some modifications to fix unit tests, etc.

LouisLandry committed:
Merge pull request #76 from ianmacl/ci

Ci

elinw committed:
Doc blocks in installer adapters

Commit:
databaesquery change from the CMS

elinw committed:
Doc block in installer adapters.

LouisLandry committed:
Merge pull request #78 from AmyStephen/master

DBQuery fix that fixes the CMS Installer's use with 11.1

elinw committed:
Some partial work

realityking committed:
Sync some changes from Joomla 1.7.

realityking committed:
Remove about half of the use of DS in paths.

This is a merge of Joomla 1.7 changes.

elkuku committed:
Strict standards: Calling a non static method statically

eddieajau committed:
Merge pull request #80 from realityking/cmsmerge

Sync some changes from Joomla 1.7.

eddieajau committed:
Sync application package with Joomla 1.7.

eddieajau committed:
Sync client package with Joomla 1.7.

eddieajau committed:
Merge branch 'master' of ssh://git@github.com/joomla/joomla-platform.git

eddieajau committed:
Trim trailing white space.

Commit:
Code Cleanup

Commit:
Fixing installer component adapter standards

Commit:
Fixing code standards - JStream

eddieajau committed:
Merge pull request #82 from ianmacl/standards

Code Cleanup

eddieajau committed:
Merge pull request #81 from elkuku/master

Enfore strict standards in JApplication::getURLSafe method (using self operator on getCfg method which is not static).

elinw committed:
Merge branch 'master' into HEAD

eddieajau committed:
Change release to 11.2 (still under development).

eddieajau committed:
Code cleaning.

eddieajau committed:
Code cleaning.

eddieajau committed:
Merge branch 'master' of ssh://git@github.com/joomla/joomla-platform.git

Conflicts:
	libraries/joomla/application/component/controller.php
	libraries/joomla/client/ldap.php

eddieajau committed:
Sync changes to Database package from Joomla 1.7.

eddieajau committed:
Merge branch 'master' of ssh://git@github.com/joomla/joomla-platform.git

eddieajau committed:
Sync changes to Error package with Joomla 1.7.

eddieajau committed:
Merge branch 'master' of ssh://git@github.com/joomla/joomla-platform.git

elinw committed:
CMS commit 21442

elinw committed:
CMS commit 21464

elinw committed:
CMS Commit 21478

eddieajau committed:
Merge pull request #83 from elinw/cmscatchup

Commit from CMS that is not in platform

elinw committed:
CMS Commit 21479

elinw committed:
CMS commit 21480

elinw committed:
CMS Commit 21490

elinw committed:
CMS commit 21533

eddieajau committed:
Merge pull request #84 from elinw/cmscatchup

Catch up commits to CMS

eddieajau committed:
Removed define for JPLATFORM from platform.php as it causes problems on some setups (probably due to slight variations in the spl autoloader).

eddieajau committed:
Merge branch 'master' of ssh://git@github.com/joomla/joomla-platform.git

elinw committed:
CMS commit 21571

eddieajau committed:
Fix JPlatform test failures.

elinw committed:
CMS Commit 21639  This is a big one

elinw committed:
CMS Commit 21645

elinw committed:
CMS Commit  21647

realityking committed:
Sync more changes from Joomla 1.7. (Mostly DS)

elinw committed:
CMS Commit 21648

elinw committed:
Commit 21649

elinw committed:
CMS Commit 21650

elinw committed:
CMS commit 21678

elinw committed:
CMS Commit 21699

elinw committed:
Fix commenting style on CMS commit 21699

elinw committed:
Typo in comment in CMS Commit 21707 (Commit already incorporated)

elinw committed:
CMS Commit 21708

elinw committed:
CMS Commit 21728

elinw committed:
CMS Commit 21744

elinw committed:
CMS Commit 21745

elinw committed:
CMS Commit 21755

elinw committed:
CMS Commit 21756

eddieajau committed:
Merge pull request #86 from elinw/cmscatchup

Catch up  commits to CMS

eddieajau commented:
Rouven, could you just resync the pull.  Elin had a commit in the queue that included some of the changes in this request.  Thanks.

eddieajau committed:
Form package docblocks sync'd with Joomla 1.7 repo.

eddieajau committed:
Registry package docblocks/code formatting sync'd with Joomla 1.7 repo.

chdemko committed:
Merge git://github.com/joomla/joomla-platform into registry

realityking committed:
Merge remote branch 'upstream/master' into cmsmerge

Conflicts:
	libraries/joomla/filesystem/archive/zip.php
	libraries/joomla/form/rules/options.php

realityking commented:
Updated.

chdemko committed:
[#26241] JCacheFIle should not use JFile calls at all

chdemko committed:
Restoring registry.php

realityking committed:
Merge the media folder.

realityking committed:
More changes from Joomla 1.7

elinw committed:
Change concat to concatenate as per mailing list discussion.

eddieajau committed:
Merge pull request #92 from elinw/databasefix

Fixes a mistake in mysql and mysqli drivers

eddieajau committed:
Fix file line endings.

eddieajau committed:
Fix line endings.

eddieajau committed:
Fix bug in JParameter bind ($group passed to loadString causing problems).

rvsjoen committed:
Make getTable fill in the prefix like JController::getView

rvsjoen committed:
Remove old workaround for webkit browsers, and make the workaround for msie more targeted

rvsjoen committed:
Backwards compatibility for 3rd party components, thanks Christophe

rvsjoen committed:
Use phputf8 library instead of multibyte extension

rvsjoen committed:
Restore konqueror redirect workaround

rvsjoen committed:
Consistent phpdoc subpackage in updater

eddieajau committed:
Merge pull request #98 from rvsjoen/phpdoc

Consistent phpdoc subpackage in updater

rvsjoen committed:
Merge remote-tracking branch 'upstream/master' into table

rvsjoen committed:
Merge remote-tracking branch 'upstream/master' into redirect

elkuku committed:
removing empty braces when instancing new a class

elkuku committed:
removing empty braces when instancing a new class

elinw committed:
Merge branch 'master' into HEAD

LouisLandry committed:
Merge pull request #43 from chdemko/registry

Improve JRegistry

LouisLandry committed:
Merge pull request #100 from elkuku/save_the_world

Code cleanup

LouisLandry committed:
Merge pull request #94 from rvsjoen/redirect

Redirect workarounds

LouisLandry committed:
Merge pull request #91 from chdemko/jcachefile

[#26241] JCacheFIle should not use JFile calls at all

LouisLandry committed:
Merge pull request #93 from rvsjoen/table

Make getTable fill in the prefix like JController::getView

elinw committed:
Doc Blocks work in application

elinw committed:
Fix some packages and small edits

elinw committed:
Fix typo

eddieajau committed:
Merge changes from Joomla CMS 1.7.

eddieajau committed:
Fixed test to add encoding propery to XML prologue.

eddieajau committed:
Changed Joomla version number in test.

eddieajau committed:
Merge changes to Filesystem package from Joomla 1.7 CMS.

mbabker committed:
Add optional context to JHtmlContent::prepare

eddieajau committed:
Merge changes to Utilities package from Joomla 1.7 CMS.

eddieajau committed:
Merge branch 'master' of ssh://git@github.com/joomla/joomla-platform.git

elinw committed:
Doc block updates in rules

eddieajau committed:
Merge cosmetic changes to Cache package from Joomla 1.7.

eddieajau committed:
Added build and test ignore files.

eddieajau committed:
Fix checkstyle warnings.

elinw committed:
Clean up in Form Fields doc blocks.

elinw committed:
More doc bloc work in form fields.

elinw committed:
File list doc blocks

elinw committed:
Merge branch 'master' into HEAD

Conflicts:
	libraries/joomla/installer/adapters/component.php
	libraries/joomla/updater/updateadapter.php

elinw committed:
More work in doc blocks in fiels

Commit:
defined('JPATH_PLATFORM') or die; was commented out.

Commit:
PHP Strict Fixes for JMail Class

elinw committed:
More doc block work in fields

mbabker committed:
Change order of variables

elinw committed:
A lot of work on JForm doc blocks

elinw committed:
Update deprecation release numbers to reflect the planned schedule
more accurately.

elinw committed:
Work on deprecation and deprecation logging

Commit:
Fixed subclass detection code

Commit:
Merging changes

Commit:
Fixing merge error

Commit:
Merge pull request #106 from ianmacl/cli

Fixed subclass detection code

Commit:
Merge pull request #101 from elinw/docblocks

Merging in docblock changes.

elinw committed:
More work on logging

elinw committed:
Merge branch 'master' into HEAD

Conflicts:
	libraries/joomla/application/router.php
	libraries/joomla/utilities/simplexml.php

Commit:
Merge pull request #105 from elinw/logging

Added logging for deprecated methods.

Commit:
Merge pull request #102 from mbabker/master

Add optional context to JHtmlContent::prepare

elinw committed:
As logging to test harness

Commit:
Merge pull request #107 from elinw/testharness

Logging and tests

elinw committed:
Missing bracket

Commit:
Merge pull request #108 from elinw/testharness

Missing bracket

Commit:
Fixing unit test issue

Commit:
Merge pull request #109 from ianmacl/utest

Fixing unit test issue

Commit:
Test

LouisLandry committed:
Merge pull request #103 from AmyStephen/master

JForm Telephone Rule

realityking committed:
Merge remote-tracking branch 'upstream/master' into cmsmerge

Conflicts:
	libraries/joomla/document/feed/renderer/rss.php
	libraries/joomla/form/rules/options.php
	libraries/joomla/installer/adapters/file.php

LouisLandry commented:
Your repository needs to be rebased according to: https://groups.google.com/d/topic/joomla-dev-framework/WpDQzYe5yRs/discussion

Our apologies but the longer we waited the worse it got.

Cheers.
